Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/database/function/VariableDB.java

```diff
@@ -389,6 +389,9 @@ public boolean isEquivalent(Variable otherVar) {
 		// If we have a VariableImpl or either function is using custom variable storage
 		// then they are only equivalent if the storage is the same.
 		Function otherFunction = otherVar.getFunction();
+		if (function != null && !(function.equals(otherFunction))) {
```

Comment:
Our function equality is identify based.  I am concerned that this change will break other uses (e.g., Merge) which does not want to restrict to the same function instance.